hstream-sql/src/HStream/SQL/Codegen/Utils.hs

```diff
@@ -148,3 +157,17 @@ diffTimeToMs diff = fromInteger $ diffTimeToPicoseconds diff `div` 10^9
 composeColName :: Maybe StreamName -> FieldName -> Text
 composeColName Nothing field       = field
 composeColName (Just stream) field = stream <> "." <> field
+
+--------------------------------------------------------------------------------
+diff :: Vector Value -> Vector Value -> Vector Value
+xs `diff` ys | null xs   = V.empty
+             | null ys   = xs
+             | otherwise = V.fromList $ V.toList xs L.\\ V.toList ys
+
+intersect :: Vector Value -> Vector Value -> Vector Value
+intersect xs ys | null xs || null ys = V.empty
+                | otherwise = V.fromList $ V.toList xs `L.intersect` V.toList ys
+
+nub :: Vector Value -> Vector Value
+nub xs | null xs   = xs
+       | otherwise = V.fromList $ L.nub (V.toList xs)
```

Comment:
> Use more specific names to describe what the function does. Also, `Vector` is NOT inductively defined.

Since `intersect` and `nub` and `Data.List` are of $O(n)$, should we examine some cases first to avoid build from and to lists?

Reply:
That is OK.

Reply:
make then consist with `Data.List` via a721102